Prevent clippy from tripping lint

kristof-mattei opened the pull request:
The following code

```rust
#[cxx_qt::bridge]
pub mod my_object {

    unsafe extern "RustQt" {
        #[qobject]
        #[qml_element]
        type Hello = super::HelloRust;
    }

    unsafe extern "RustQt" {
        #[qinvokable]
        pub fn say_hello(self: &Hello);
    }
}

#[derive(Default)]
pub struct HelloRust {}

impl my_object::Hello {
    pub fn say_hello(&self) {
        println!("Hello world!");
    }
}
```

Triggers 

```
❮ cargo clippy -- -W clippy::pedantic
   Compiling qt-test v0.0.0-development (/Users/kristof/sources/kristof-mattei/qt-test)
warning: boxed return of the sized type `cxxqt_object::HelloRust`
 --> src/cxxqt_object.rs:1:1
  |
1 | #[cxx_qt::bridge]
  | ^^^^^^^^^^^^^^^^^
  |
  = help: changing this also requires a change to the return expressions in this function
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_box_returns
  = note: `-W clippy::unnecessary-box-returns` implied by `-W clippy::pedantic`
  = help: to override `-W clippy::pedantic` add `#[allow(clippy::unnecessary_box_returns)]`
  = note: this warning originates in the attribute macro `cxx_qt::bridge` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: `qt-test` (bin "qt-test") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 11.27s
```

I have `clippy::pedantic` on and disable the ones I don't need, but I cannot disable this inside of the macro.

And since the macro always returns a `Box` I don't believe there is any harm in allowing it there.

ahayzen-kdab commented:
Thanks! This looks good, i assume there aren't any other boxes we have that also trigger this :thinking: And note that the CI failures are unrelated to this branch and are regressions happening on main too.

kristof-mattei commented:
> Thanks! This looks good, i assume there aren't any other boxes we have that also trigger this 🤔 And note that the CI failures are unrelated to this branch and are regressions happening on main too.

Went through the code base, updated the other instance where an `std::boxed::Box` is returned, and updated the expected test outputs. 

Equally, ran tests and no errors returned contained either `boxed` or `unnecessary_box_returns `, to ensure I actually covered all cases.

ahayzen-kdab reviewed (approved):
LGTM, thanks!
